gamestonk_terminal/economy/fred_view.py

```diff
@@ -1,194 +1,131 @@
-                    (p[success],) = axes.plot(
-                        df_fred.index,
-                        df_fred.iloc[:, 0],
-                        c=l_colors[success],
-                        label=series_id.upper(),
-                    )
-
-                    axes.yaxis.label.set_color(l_colors[success])
-
-                    l_ts_start.append(df_fred.index[0])
-                    l_ts_end.append(df_fred.index[-1])
-
-            if not ns_parser.text and success:
-                plt.title("FRED: " + ", ".join(success_series))
-                plt.xlim(min(l_ts_start), max(l_ts_end))
-                plt.gcf().autofmt_xdate()
-                plt.xlabel("Time")
-                plt.legend([val for _, val in p.items()], success_series)
-                plt.gca().spines["left"].set_visible(False)
-                plt.show()
-            print("")
-
-        else:
-            d_data = fred.get_series(ns_parser.series_id, ns_parser.start_date)
-
-            df_fred = pd.DataFrame(d_data, columns=[f"{ns_parser.series_id}"])
             df_fred.index.name = "Date"
 
-            if ns_parser.text:
-                print(df_fred.to_string())
+            if raw:
+                print(ser)
+                print("-" * len(ser))
+                print(df_fred.dropna().to_string(), "\n")
 
             else:
-                plt.figure(figsize=plot_autoscale(), dpi=PLOT_DPI)
-                plt.plot(df_fred.index, df_fred.iloc[:, 0])
-                plt.xlabel("Time")
-                plt.xlim(df_fred.index[0], df_fred.index[-1])
-                plt.ylabel(f"{ns_parser.series_id.upper()}")
-                plt.grid(b=True, which="major", color="#666666", linestyle="-")
-                plt.minorticks_on()
-                plt.grid(
-                    b=True, which="minor", color="#999999", linestyle="-", alpha=0.2
+                success += 1
+                success_series.append(ser.upper())
+
+                axes = ax.twinx()
+                axes.spines["right"].set_position(("axes", 1 + success * 0.15))
+                axes.spines["right"].set_color(l_colors[success])
+                (p[success],) = axes.plot(
+                    df_fred.index,
+                    df_fred.values,
+                    c=l_colors[success],
+                    label=ser.upper(),
                 )
-                if choice:
-                    plt.title(fred_series_description[ns_parser.series_id])
-                if gtff.USE_ION:
-                    plt.ion()
-                plt.show()
-            print("")
-
-    except SystemExit:
-        print("")
-    except Exception as e:
-        print(e, "\n")
-        return
+
+                axes.yaxis.label.set_color(l_colors[success])
+
+                l_ts_start.append(df_fred.index[0])
+                l_ts_end.append(df_fred.index[-1])
+
+    if not raw and success > -1:
+        plt.title("FRED: " + ", ".join(success_series))
```

Comment:
As I mentioned in one of the comments, is there a command with the fred api that could return a useful title?

Reply:
Actually there is. Let me try to refactor code to incorporate this. I'll try to put it in a legend

Reply:
![image](https://user-images.githubusercontent.com/25267873/128644582-1ff6ed7c-80f2-4480-a92b-9fd91381833c.png)

Reply:
i like that